java/fury-core/src/main/java/org/apache/fury/serializer/NonexistentClassSerializers.java

```diff
@@ -241,13 +252,40 @@ public Object read(MemoryBuffer buffer) {
       return enumConstants[ordinal];
     }
   }
+    public static final class NonexistentEnumByNameClassSerializer extends Serializer {
+        private final Map<MetaStringBytes, NonexistentEnum> enumMap;
+        private final MetaStringResolver metaStringResolver;
+
+        public NonexistentEnumByNameClassSerializer(Fury fury) {
+            super(fury, NonexistentEnum.class);
+            metaStringResolver = new MetaStringResolver();
+            NonexistentEnum[] enumConstants = NonexistentEnum.class.getEnumConstants();
+            enumMap = Arrays.stream(enumConstants)
+                    .map(e -> {
+                        MetaStringBytes stringBytes =
+                                this.metaStringResolver.getOrCreateMetaStringBytes(GENERIC_ENCODER.encode(e.name(), MetaString.Encoding.UTF_8));
+                        return new AbstractMap.SimpleEntry<>(stringBytes, e);
+                    })
+                    .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+        }
 
+        @Override
+        public Enum read(MemoryBuffer buffer) {
+            MetaStringBytes metaStringBytes = metaStringResolver.readMetaStringBytes(buffer);
+            if (!enumMap.containsKey(metaStringBytes)) {
+                Optional<NonexistentEnum> first = enumMap.values().stream().findFirst();
+                return first.orElse(null);
+            }
+            return enumMap.get(metaStringBytes);
```

Comment:
This introduce a second map loopkup, we should avoid it. we can `value = enumMap.get(xxx);` and check whether `value` is null instead of check by `containsKey`